lib/supavisor/helpers.ex

```diff
@@ -4,16 +4,27 @@ defmodule Supavisor.Helpers do
   @spec check_creds_get_ver(map) :: {:ok, String.t()} | {:error, String.t()}
   def check_creds_get_ver(params) do
     Enum.reduce_while(params["users"], {nil, nil}, fn user, _ ->
+      ssl_opts =
+        if params["upstream_ssl"] and params["upstream_verify"] == "peer" do
```

Comment:
would `params["upstream_ssl"]` be `true` or `"true"` (string)? The latter would crash `and`. I think it's better it's `true` and we use `and` but just double-checking!

Reply:
the `upstream_ssl` is a boolean, but it's not required, so it can also be `nil`. I've added double negation
```elixir
if !!params["upstream_ssl"] and params["upstream_verify"] == "peer" do
```